rocAL/rocAL_pybind/example/coco_pipeline.py

```diff
@@ -197,24 +230,36 @@ def __next__(self):
                     actual_labels.append(encodded_labels_tensor[i][idx].tolist())
 
             if self.display:
-               img = torch.from_numpy(self.out)
-               draw_patches(img[i], self.image_id[i], actual_bboxes)
+                if self.device == "cpu":
+                    img = torch.from_numpy(self.out)
+                else:
+                    dlpack_array = self.hip_array.toDlpack()
+                    img = dp.from_dlpack(dlpack_array)
+                draw_patches(img[i], self.image_id[i], actual_bboxes, self.device)
 
-        if self.tensor_dtype == types.FLOAT:
-            return torch.from_numpy(self.out), encoded_bboxes_tensor, encodded_labels_tensor, image_id_tensor, image_size_tensor
-        elif self.tensor_dtype == types.FLOAT16:
-            return torch.from_numpy(self.out.astype(np.float16)), encoded_bboxes_tensor, encodded_labels_tensor, image_id_tensor, image_size_tensor
+        if self.device == "cpu":
+            if self.tensor_dtype == types.FLOAT:
+                return torch.from_numpy(self.out), encoded_bboxes_tensor, encodded_labels_tensor, image_id_tensor, image_size_tensor
+            elif self.tensor_dtype == types.FLOAT16:
+                return torch.from_numpy(self.out.astype(np.float16)), encoded_bboxes_tensor, encodded_labels_tensor, image_id_tensor, image_size_tensor
+
+        if self.device == "cuda":
+            dlpack_array = self.hip_array.toDlpack()
```

Comment:
Eventually we need to get rid of all these calls and do the conversion to numpy in the pybind wrapper function.